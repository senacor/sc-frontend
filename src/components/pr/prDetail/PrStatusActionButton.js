import React from 'react';
import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles/index';

const styles = theme => ({
  buttonDesktop: {
    position: 'relative',
    backgroundColor: theme.palette.primary['400'],
    color: '#FFF',
    marginBottom: '2%'
  },
  sizeSmall: {}
});

function PrStatusActionButton(props) {
  let { classes, label } = props;
  return (
    <Button
      classes={{ sizeSmall: classes.sizeSmall }}
      size="small"
      className={classes.buttonDesktop}
      onClick={props.releaseButtonClick}
    >
      {label}
    </Button>
  );
}

export default withStyles(styles)(PrStatusActionButton);
