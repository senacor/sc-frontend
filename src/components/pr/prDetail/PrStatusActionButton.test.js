import React from 'react';
import PrStatusActionButton from './PrStatusActionButton';
import { createShallow } from '@material-ui/core/test-utils';

describe('PerformanceReviewDetail Component', () => {
  let shallow = createShallow({ dive: true });

  const mockReleaseButtonClick = jest.fn();

  it('should match snapshot', () => {
    const component = shallow(
      <PrStatusActionButton
        label={'Freigabe'}
        releaseButtonClick={mockReleaseButtonClick}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
