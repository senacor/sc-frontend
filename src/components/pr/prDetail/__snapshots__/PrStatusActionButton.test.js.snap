// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PerformanceReviewDetail Component should match snapshot 1`] = `
<WithStyles(Button)
  className="PrStatusActionButton-buttonDesktop-1"
  classes={
    Object {
      "sizeSmall": "PrStatusActionButton-sizeSmall-2",
    }
  }
  onClick={[Function]}
  size="small"
>
  Freigabe
</WithStyles(Button)>
`;
